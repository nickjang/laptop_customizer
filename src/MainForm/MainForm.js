import React, { Component } from 'react';
import Specs from '../Specs/Specs';

import './MainForm.css'

class CustomizerForm extends Component {
  static defaultProps = {
    features: {},
    selected: {} ,
    updateFeature: () => {}
  }
  
  render() {
    const features = Object.keys(this.props.features).map((feature, idx) => {
      const featureHash = feature + '-' + idx;
      return (
        <Specs key={featureHash} idx={idx} featureName={feature}
          feature={this.props.features[feature]} updateFeature={this.props.updateFeature}
          selectedFeatureName={this.props.selected[feature].name} />
      );
    });

    return (
      <form className="main__form">
        <h2>Customize your laptop</h2>
        {features}
      </form>
    );
  }
}

export default CustomizerForm;