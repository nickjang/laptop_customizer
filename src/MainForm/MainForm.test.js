import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainForm from './MainForm';
import FEATURES from '../FEATURES';

const selected = {
  Processor: {
    name: '17th Generation Intel Core HB (7 Core with donut spare)',
    cost: 700
  },
  'Operating System': {
    name: 'Ubuntu Linux 16.04',
    cost: 200
  },
  'Video Card': {
    name: 'Toyota Corolla 1.5v',
    cost: 1150.98
  },
  Display: {
    name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
    cost: 1500
  }
}

describe('<MainForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MainForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders empty when no props supplied', () => {
    const wrapper = shallow(<MainForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders form with features when props are supplied', () => {
    const wrapper = shallow(
      <MainForm features={FEATURES} selected={selected} 
            updateFeature={() => {}} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});