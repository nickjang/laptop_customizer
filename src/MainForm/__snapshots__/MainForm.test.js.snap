// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainForm /> renders empty when no props supplied 1`] = `
<form
  className="main__form"
>
  <h2>
    Customize your laptop
  </h2>
</form>
`;

exports[`<MainForm /> renders form with features when props are supplied 1`] = `
<form
  className="main__form"
>
  <h2>
    Customize your laptop
  </h2>
  <Specs
    feature={
      Array [
        Object {
          "cost": 700,
          "name": "17th Generation Intel Core HB (7 Core with donut spare)",
        },
        Object {
          "cost": 1200,
          "name": "Professor X AMD Fire Breather with sidewinder technology",
        },
      ]
    }
    featureName="Processor"
    idx={0}
    key="Processor-0"
    selectedFeatureName="17th Generation Intel Core HB (7 Core with donut spare)"
    updateFeature={[Function]}
  />
  <Specs
    feature={
      Array [
        Object {
          "cost": 200,
          "name": "Ubuntu Linux 16.04",
        },
        Object {
          "cost": 300,
          "name": "Bodhi Linux",
        },
      ]
    }
    featureName="Operating System"
    idx={1}
    key="Operating System-1"
    selectedFeatureName="Ubuntu Linux 16.04"
    updateFeature={[Function]}
  />
  <Specs
    feature={
      Array [
        Object {
          "cost": 1150.98,
          "name": "Toyota Corolla 1.5v",
        },
        Object {
          "cost": 1345,
          "name": "Mind mild breeze 2000",
        },
      ]
    }
    featureName="Video Card"
    idx={2}
    key="Video Card-2"
    selectedFeatureName="Toyota Corolla 1.5v"
    updateFeature={[Function]}
  />
  <Specs
    feature={
      Array [
        Object {
          "cost": 1500,
          "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
        },
        Object {
          "cost": 1400,
          "name": "15.3\\" HGTV (3840 x 2160) Home makeover edition",
        },
      ]
    }
    featureName="Display"
    idx={3}
    key="Display-3"
    selectedFeatureName="15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs"
    updateFeature={[Function]}
  />
</form>
`;
